test/check-machines-migrate

```diff
@@ -0,0 +1,231 @@
+        # Equivalent storage pool is not present on the destination
+        self._testMigrationGeneric(True, False, "pool")
+
+    # Only qemu 5.1.0 and above does checking for volumes compatibility during migration
+    # See https://mail.gnu.org/archive/html/qemu-devel/2020-04/msg05559.html
+    @skipImage('Fails validation because ubuntu-2004 with qemu version below 5.1.0 doesnt compare volume sizes', "ubuntu-2004")
+    def testFailMigrationVolumesNotCompatible(self):
+        # Different sized volume already exists on destination
+        self._testMigrationGeneric(True, False, "volume")
+
+    def testFailMigrationTCPPortClosed(self):
+        # Ports for TCP tranfer are closed
+        self._testMigrationGeneric(True, False, "port_closed")
+
+    def testFailMigrationAuthentication(self):
+        # No key authentication is set up
+        self._testMigrationGeneric(True, False, "authentication")
+
+    def testFailMigrationUriIncorrect(self):
+        # Incorrect uri inputted
+        self._testMigrationGeneric(True, False, "uri")
+
+    def testFailMigrationDomainUnknown(self):
+        # Different sized volume already exists on destination
+        self._testMigrationGeneric(True, False, "domain")
+
+    def testFailMigrationStopLibvirtd(self):
+        # Libvirtd.service is not working on the destination
+        self._testMigrationGeneric(True, False, "libvirtd")
+
+    def testFailMigrationKillDestination(self):
+        # Kill destination server in the middle of migration
+        self._testMigrationGeneric(True, False, "kill")
+
+    def _testMigrationGeneric(self, copy_storage, temporary, fail):
+        b = self.browser
+        machine1 = self.machine
+        machine2 = self.machines['machine2']
+
+        self.createVm("subVmTest1")
+        self.login_and_go("/machines")
+        self.prepare(machine1, machine2, fail)
+
+        if not copy_storage:
+            self.preCreateStorageOnDest(machine1, machine2)
+        # Older libvirt needs the destination storage preallocated
+        elif self.machine.image in ["ubuntu-2004"] and fail != "pool":
+            machine2.execute("qemu-img create -f qcow2 /var/lib/libvirt/images/subVmTest1-2.img 50M")
+
+        self.waitVmRow("subVmTest1")
+        self.goToVmPage("subVmTest1")
+
+        self.performAction("subVmTest1", "migrate")
+        b.wait_visible("#migrate-modal")
+
+        if fail == "uri":
+            # test failure for syntactically incorrect VM
+            b.set_input_text("#dest-uri-input", "qemu+ssh://root@machine2/sistem")
+        elif fail == "domain":
+            # test failure for unkwnown domain
+            b.set_input_text("#dest-uri-input", "qemu+ssh://root@someunknowndomain/system")
+        else:
+            b.set_input_text("#dest-uri-input", "qemu+ssh://root@machine2/system")
+
+        if copy_storage:
+            b.click("#copy")
+        else:
+            b.click("#shared")
+
+        b.wait_visible("#temporary")
```

Comment:
(Not necessary, set_checked() and all our other test API wait for the selector already.)